### ISSUE TITLE
Stop shipping Desktop OOB package for System.Runtime.Serialization.Primitives

### ISSUE THREAD

**Issue by karelz** [dotnet/corefx#16804]
Shipping OOB packages overriding Desktop .NET Framework (gac'd assemblies) is HARD.
We ran into problems with Networking and decided to stop shipping all such packages.

We should stop shipping System.Runtime.Serialization.Primitives package for Desktop and forward everything to Desktop .NET Framework.

@shmao @zhenlan is that feasible? What will we lose if we stop shipping the OOB package for Desktop? Some APIs? Bug fixes? Anything super-important?

cc @weshaggard

**Comment by zhenlan** [comment]
System.Runtime.Serialization.Primitives is a partial facade assembly to desktop .NET. Theoretical, apps using `ISerializationSurrogateProvider` on full .NET will be broken if we stop shipping it.

https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Serialization.Primitives/src/System.Runtime.Serialization.Primitives.csproj#L41

**Comment by karelz** [comment]
How much is that interface used, what will break? (note: [apisof.net](https://apisof.net/catalog/System.Runtime.Serialization.ISerializationSurrogateProvider) shows no usage on NuGet)
Note that we intentionally broke ~100 APIs in .NET Standard 2.0 on Desktop - they will throw MissingMethod/MissingType exceptions. Most of them, because we can't do anything else in .NET Standard, but 8 of them (System.Net.Http #11100/#16805 and System.IO.Compression #15693), because exactly the same reason as listed here.

The trouble is that shipping Desktop inbox replacements is HARD - package authors are not typically aware of all the implications. There are also nasty tooling side effects in the end-to-end story - see #9846. And there is no test coverage from CoreFX infra.

@zhenlan if you decide to keep the package as is, it would be best to sync up with me, @weshaggard and @ericstj for your team to fully understand what you are shipping, what kind of testing you are expected to do, etc. (note: none of our existing infra tests the scenario, so it is up to manual/ad-hoc testing)

**Comment by weshaggard** [comment]
>Also, can you please confirm if this is the last OOB package in CoreFX that overrides Desktop? Or do we have more?

Answering your other question @karelz. I've not done any analysis to confirm or deny that, as we talked in the past someone would need to go through all our packages and look at what they produce and compare it to what is inbox. @ericstj and I called out HttpClient and IO Compression as 2 examples that we know we are overwriting implementation that was inbox but there might be other partial facades that add stuff on top of the desktop.
